nilearn/_utils/niimg_conversions.py

```diff
@@ -135,7 +162,10 @@ def concat_niimgs(niimgs, dtype=np.float32, accept_4d=False,
     """
 
     # get properties from first image
-    first_niimg = check_niimg(next(iter(niimgs)))
+    try:
+        first_niimg = check_niimg(next(iter(niimgs)), atleast_4d=True)
+    except StopIteration:
+        raise TypeError('Cannot concatenate empty objects')
```

Comment:
maybe ValueError rather than TypeError, since an empty list still has the right type ?